qemu/tests/create_external_snapshot_via_driver.py

```diff
@@ -0,0 +1,135 @@
+
+        """
+        Prepare ssh environment
+        :return: None
+        """
+        error_context.context("STEP 1: Prepare ssh login environment.")
+        clean_key()
+        copy_pub_key_cmd = params.get("copy_pub_key_cmd")
+        logging.info("Create ssh key & copy it using ssh-copy-id")
+        process.system(params["create_ssh_key_cmd"], shell=True)
+        logging.info("Copy the ssh public key to remote host")
+        process.system(copy_pub_key_cmd, shell=True)
+
+    @error_context.context_aware
+    def create_https_test():
+
+        """
+        Prepare https environment
+        :return: None
+        """
+        error_context.context("STEP 1: Prepare https login environment.")
+        process.system("yum install -y mod_ssl")
+        logging.info("Start httpd service")
+        process.system(params.get("http_conf_cmd"), shell=True)
+        process.system(params.get("httpd_start_cmd"), shell=True)
+        out = process.system_output(params.get("httpd_status_cmd"), shell=True)
+        if not re.search(params.get("httpd_status_re"), out):
+            err = "Http server fails to start, because of {}".format(out)
+            exceptions.TestError(err)
+
+    subcommand = params.get("subcommand")
+    eval("%s_test()" % subcommand)
+
+    user_profile = ""
+    user_profile_backup = ""
+    if params.get("external_snapshot_protocol") == "ssh":
+        user_profile = params.get("user_profile")
+        user_profile_backup = params.get("user_profile_backup")
+        shutil.copy(user_profile, user_profile_backup)
+
+    base_image = params.get("images", "image1").split()[0]
+    params.update(
+        {"image_name_%s" % base_image: params["image_name"],
+         "image_format_%s" % base_image: params["image_format"]})
+    t_file = params["guest_file_name_%s" % base_image]
+    snapshot_test = qemu_disk_img_info.InfoTest(
+        test, params, env, base_image)
+    logging.info("STEP 2: Save file md5sum before create snapshot.")
+    snapshot_test.start_vm(params)
+    md5_before_snapshot = snapshot_test.save_file(t_file)
+    if not md5_before_snapshot:
+        raise exceptions.TestError("Fail to save tmp file.")
+    snapshot_test.destroy_vm()
+
+    logging.info("STEP 3: Start to create snapshot")
+    snapshot_test_2 = qemu_disk_img_info.InfoTest(test, params, env, "sn1")
+    n_params = snapshot_test_2.create_snapshot()
+    if params.get("external_snapshot_protocol") == "ssh":
+        ssh_cfg = process.system_output("cat /root/.bash_profile", shell=True)
+        ssh_sock = re.search("(SSH_AUTH_SOCK)=(.*?);", ssh_cfg)
+        ssh_pid = re.search("(SSH_AGENT_PID)=(\d+);", ssh_cfg)
+        command_prefix = "%s=%s;export %s;"
+        command_prefix += "%s=%s;export %s;"
+        ssh_auth_sock = ssh_sock.group(1)
+        ssh_sock_path = ssh_sock.group(2)
+        ssh_agent_pid = ssh_pid.group(1)
+        ssh_pid_num = ssh_pid.group(2)
+
+        n_params["qemu_command_prefix"] = command_prefix % (
```

Comment:
As you redirect SSH_AUTH_SOCK and SSH_AGENT_PID to /root/.bash_profile, can we simply make "qemu_command_prefix" = "source ${user_profile}"